integration_tests/src/main/python/date_time_test.py

```diff
@@ -558,3 +559,87 @@ def test_timestamp_millis_long_overflow():
 def test_timestamp_micros(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark : unary_op_df(spark, data_gen).selectExpr("timestamp_micros(a)"))
+
+
+# used by timezone test cases
+def get_timezone_df(spark):
+    schema = StructType([
+        StructField("ts_str_col", StringType()),
+        StructField("long_col", LongType()),
+        StructField("ts_col", TimestampType()),
+        StructField("date_col", DateType()),
+        StructField("date_str_col", StringType()),
+    ])
+    data = [
+        ('1970-01-01 00:00:00', 0, datetime(1971, 1, 1), date(1971, 1, 1), '1971-01-01'),
+        ('1970-01-01 00:00:00', 0, datetime(1971, 1, 1), date(1971, 1, 1), '1971-01-01'),
+    ]
+    return spark.createDataFrame(SparkContext.getOrCreate().parallelize(data),schema)
+
+# used by timezone test cases, specify all the sqls that will be impacted by non-utc timezone
+time_zone_sql_conf_pairs = [
+    ("select minute(ts_col) from tab", {}),
+    ("select second(ts_col) from tab", {}),
+    ("select hour(ts_col) from tab", {}),
+    ("select date_col + (interval 10 days 3 seconds) from tab", {}),
+    ("select date_format(ts_col, 'yyyy-MM-dd HH:mm:ss') from tab", {}),
+    ("select unix_timestamp(ts_col) from tab", {"spark.rapids.sql.improvedTimeOps.enabled": "true"}),
+    ("select to_unix_timestamp(ts_str_col) from tab", {"spark.rapids.sql.improvedTimeOps.enabled": "false"}),
+    ("select to_unix_timestamp(ts_col) from tab", {"spark.rapids.sql.improvedTimeOps.enabled": "true"}), 
+    ("select to_date(date_str_col, 'yyyy-MM-dd') from tab", {}), # test GpuGetTimestamp
+    ("select to_date(date_str_col) from tab", {}),
+    ("select from_unixtime(long_col, 'yyyy-MM-dd HH:mm:ss') from tab", {}),
+    ("select cast(ts_col as string) from tab", {}), # cast
+    ("select cast(ts_col as date) from tab", {}), # cast
+    ("select cast(date_col as TIMESTAMP) from tab", {}), # cast
+    ("select to_timestamp(ts_str_col) from tab", {"spark.rapids.sql.improvedTimeOps.enabled": "false"}), 
+    ("select to_timestamp(ts_str_col) from tab", {"spark.rapids.sql.improvedTimeOps.enabled": "true"}), 
+    ]
+
+
+@allow_non_gpu("ProjectExec")
+@pytest.mark.parametrize('sql, extra_conf', time_zone_sql_conf_pairs)
+def test_timezone_for_operators_with_non_utc(sql, extra_conf):
+    # timezone is non-utc, should fallback to CPU
+    timezone_conf = {"spark.sql.session.timeZone": "+08:00", 
```

Comment:
Should we make the time zone string a param to the test? Just because I would like to test a few more time zones than just `+08:00`

Reply:
Done